assets/data-port/import/data/reducer.js

```diff
@@ -149,22 +157,51 @@ export default ( state = DEFAULT_STATE, action ) => {
 				},
 				action.level,
 				{
+					...state.upload[ action.level ],
 					...action.data.upload[ action.level ],
-					inProgress: false,
+					isUploading: false,
+					isDeleting: false,
 					hasError: false,
 					errorMsg: null,
 				}
 			);
 
 		case ERROR_UPLOAD_IMPORT_DATA_FILE:
 			return updateLevelState( state, action.level, {
+				...state.upload[ action.level ],
 				isUploaded: false,
-				inProgress: false,
+				isUploading: false,
+				isDeleting: false,
 				hasError: true,
 				errorMsg: action.error.message,
 				filename: null,
 			} );
 
+		case START_DELETE_IMPORT_DATA_FILE:
+			return updateLevelState( state, action.level, {
+				...state.upload[ action.level ],
+				isDeleting: true,
+			} );
+
+		case SUCCESS_DELETE_IMPORT_DATA_FILE:
+			return updateLevelState( state, action.level, {
+				...action.data.upload[ action.level ],
+				isUploaded: false,
+				isDeleting: false,
+				hasError: false,
+				errorMsg: null,
+				filename: null,
```

Comment:
I think we have to move these to `START_DELETE_IMPORT_DATA_FILE` to clear the state as soon as they click delete. For one, it's a bit sluggish currently with only showing feedback after the server is done, but the bigger issue is, deleting a file and quickly clicking Continue will cause the job to get stuck on the progress step.

Reply:
What should it do on error? Do we need to store the previous state elsewhere?

Reply:
Just blocking `Continue` then sounds better, so the user can still retry deleting.
For the feedback, a `<Spinner>` could work in place of the delete button while it's in progress, or a `Removing..` label on the upload button?

Reply:
Blocked `Continue` be4d71396ebb077eccfe0d826b03f6c885563273
Switched to `<Spinner />` while deleting 0dae4485c55fa0f6aa782e777515860c74651837